lib/widgets/resources/arm.dart

```diff
@@ -0,0 +1,168 @@
+import 'package:collection/collection.dart';
+import 'package:flutter/material.dart';
+import 'package:viam_sdk/viam_sdk.dart';
+import 'package:viam_sdk/widgets.dart';
+
+/// A widget to control an [Arm].
+class ViamArmWidget extends StatefulWidget {
+  /// The [Arm]
+  final Arm arm;
+
+  const ViamArmWidget({
+    Key? key,
+    required this.arm,
+  }) : super(key: key);
+
+  @override
+  State<ViamArmWidget> createState() => _ViamArmWidgetState();
+}
+
+enum _PoseField { x, y, z, theta, oX, oY, oZ }
+
+class _ViamArmWidgetState extends State<ViamArmWidget> {
+  Pose endPosition = Pose();
+  List<double> jointPositions = [];
+
+  Future<void> _getPositions() async {
+    final ep = await widget.arm.endPosition();
+    final jp = await widget.arm.jointPositions();
+    setState(() {
+      jointPositions = jp;
+      endPosition = ep;
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _getPositions();
+  }
+
+  Future<void> updateEndPosition(_PoseField field, double increment) async {
+    final ep = endPosition;
+    switch (field) {
+      case _PoseField.x:
+        ep.x += increment;
+      case _PoseField.y:
+        ep.y += increment;
+      case _PoseField.z:
+        ep.z += increment;
+      case _PoseField.theta:
+        ep.theta += increment;
+      case _PoseField.oX:
+        ep.oX += increment;
+      case _PoseField.oY:
+        ep.oY += increment;
+      case _PoseField.oZ:
+        ep.oZ += increment;
+    }
+
+    await widget.arm.moveToPosition(ep);
+    await _getPositions();
+  }
+
+  Future<void> updateJointPosition(int joint, double increment) async {
+    final jp = jointPositions;
+    jp[joint] += increment;
+    await widget.arm.moveToJointPositions(jp);
+    await _getPositions();
+  }
+
+  TableRow _getEndPositionRow(_PoseField field) {
```

Comment:
Nice! Can we put this in it's own widget instead of a helper function? See [here](https://steveos.medium.com/use-separate-widget-over-helper-method-in-flutter-heres-why-better-performace-433672eb7461) for a bit more of a deep dive on why. Basically it's more performant when the widget tree rebuilds.

Reply:
@njooma and I spoke offline, I was mistaken and TableRow is not a `Widget` (<--- WHAT?!) so I was mistaken. this helper function in this case is fine :)